pkg/scheduler/cache/usagedb/prometheus/prometheus.go

```diff
@@ -145,3 +166,59 @@ func getExponentialDecayQuery(halfLifePeriod *time.Duration) string {
 
 	return fmt.Sprintf("0.5^((%d - time()) / %f)", now, halfLifeSeconds)
 }
+
+func (p *PrometheusClient) queryTumblingWindowUsage(ctx context.Context, allocationMetric string) (map[common_info.QueueID]float64, error) {
+	queuesUsage := make(map[common_info.QueueID]float64)
+
+	usageQuery := fmt.Sprintf("(%s)[%s:%s]",
+		allocationMetric,
+		p.usageParams.WindowSize.String(),
+		p.queryResolution.String(),
+	)
+
+	usageResult, warnings, err := p.client.Query(ctx, usageQuery, time.Now())
```

Comment:
Looks like a lot of code is duplicated - can you not just change the query string? All the error handling and data processing should look the same